dask/array/tests/test_rechunk.py

```diff
@@ -393,3 +393,163 @@ def test_dont_concatenate_single_chunks(shape, chunks):
+    expected = [
+        (((0, slice(0, None, None)), (0, slice(0, 5, None))),
+         ((1, slice(0, None, None)), (0, slice(0, 5, None)))),
+        (((0, slice(0, None, None)), (0, slice(5, 10, None))),
+         ((1, slice(0, None, None)), (0, slice(5, 10, None)))),
+        (((1, slice(0, None, None)), (0, slice(0, 5, None))),),
+        (((1, slice(0, None, None)), (0, slice(5, 10, None))),),
+        (((1, slice(0, None, None)), (0, slice(0, 5, None))),),
+        (((1, slice(0, None, None)), (0, slice(5, 10, None))),),
+        (((1, slice(0, None, None)), (0, slice(0, 5, None))),),
+        (((1, slice(0, None, None)), (0, slice(5, 10, None))),)]
+    assert result == expected
+
+
+def test_rechunk_unknown_from_pandas():
+    dd = pytest.importorskip('dask.dataframe')
+    pd = pytest.importorskip('pandas')
+
+    x = dd.from_pandas(pd.DataFrame(np.random.randn(50, 10)), 2).values
+    # result = x.rechunk({1: 5})
+    result = x.rechunk((None, (5, 5)))
+    assert np.isnan(x.chunks[0]).all()
+    assert np.isnan(result.chunks[0]).all()
+    assert result.chunks[1] == (5, 5)
+
+
+def test_rechunk_unknown_from_array():
+    dd = pytest.importorskip('dask.dataframe')
+    # pd = pytest.importorskip('pandas')
+    x = dd.from_array(da.ones(shape=(4, 4), chunks=(2, 2))).values
+    # result = x.rechunk({1: 5})
+    result = x.rechunk((None, 4))
+    assert np.isnan(x.chunks[0]).all()
+    assert np.isnan(result.chunks[0]).all()
+    assert x.chunks[1] == (4,)
+
+
+@pytest.mark.parametrize('x, chunks', [
+    (da.ones(shape=(50, 10), chunks=(25, 10)), (None, 5)),
+    (da.ones(shape=(50, 10), chunks=(25, 10)), {1: 5}),
+    (da.ones(shape=(50, 10), chunks=(25, 10)), (None, (5, 5))),
+
+    (da.ones(shape=(1000, 10), chunks=(5, 10)), (None, 5)),
+    (da.ones(shape=(1000, 10), chunks=(5, 10)), {1: 5}),
+    (da.ones(shape=(1000, 10), chunks=(5, 10)), (None, (5, 5))),
+
+    (da.ones(shape=(10, 10), chunks=(10, 10)), (None, 5)),
+    (da.ones(shape=(10, 10), chunks=(10, 10)), {1: 5}),
+    (da.ones(shape=(10, 10), chunks=(10, 10)), (None, (5, 5))),
+
+    (da.ones(shape=(10, 10), chunks=(10, 2)), (None, 5)),
+    (da.ones(shape=(10, 10), chunks=(10, 2)), {1: 5}),
+    (da.ones(shape=(10, 10), chunks=(10, 2)), (None, (5, 5))),
+])
+def test_rechunk_unknown(x, chunks):
+    dd = pytest.importorskip('dask.dataframe')
+    y = dd.from_array(x).values
+    result = y.rechunk(chunks)
+    expected = x.rechunk(chunks)
+
+    assert_chunks_match(result.chunks, expected.chunks)
+
+
+def assert_chunks_match(left, right):
+    for x, y in zip(left, right):
+        if np.isnan(x).any():
+            assert np.isnan(x).all()
+        else:
+            assert x == y
+
+
+def test_rechunk_unknown_raises():
+    dd = pytest.importorskip('dask.dataframe')
+
+    x = dd.from_array(da.ones(shape=(10, 10), chunks=(5, 5))).values
+    with pytest.raises(ValueError):
+        x.rechunk((None, (5, 5, 5)))
```

Comment:
Thin this case, the `ValueError` comes from the 1-dimension chunks not adding up (10 -> 15). I do have tests ensuring that `None` works as a dummy for "unchanged", though I see now that I don't have a test like `x.rechunk(((nan, nan), (5, 5)))`, which should work, and `x.rechunk(((nan,), (5, 5)))`, which should fail. I'll add them now.